src/parser/warlock/destruction/modules/core/SpellUsable.js

```diff
@@ -0,0 +1,30 @@
+import CoreSpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELLS from 'common/SPELLS';
+
+/*
+  Soul Fire (Tier 15 Destruction talent):
+    Burns the enemy's soul, dealing X Fire damage. Cooldown is reduced by 2 sec for every Soul Shard you spend.
+ */
+const REDUCTION_MS_PER_SHARD = 2000;
+
+class SpellUsable extends CoreSpellUsable {
+  hasSF = false;
+
+  constructor(...args) {
+    super(...args);
+    this.hasSF = this.selectedCombatant.hasTalent(SPELLS.SOUL_FIRE_TALENT.id);
+  }
+
+  on_byPlayer_spendresource(event) {
+    if (!this.hasSF) {
+      return;
+    }
+    if (this.isOnCooldown(SPELLS.SOUL_FIRE_TALENT.id)) {
+      // event.resourceChange is in multiples of 10 (2 shards = 20)
+      const shardsSpent = event.resourceChange / 10;
+      this.reduceCooldown(SPELLS.SOUL_FIRE_TALENT.id, shardsSpent * REDUCTION_MS_PER_SHARD);
+    }
+  }
```

Comment:
Just confirming here - shards **only** show in multiples of 10 in logs?

Reply:
I can check a few logs to be sure but afaik yes, 20 or 30.

Reply:
I'm fairly sure it is, but once you've checked to be sure you can hit the merge button.

Reply:
Looked through 3 logs with Soul Fire (smh, only after 1000th place on LFR Taloc), all spendresource events had resourceChange in multiples of 10.